pex/compatibility.py

```diff
@@ -28,6 +30,11 @@
 unicode_string = (str,) if PY3 else (unicode,)
 bytes = (bytes,)
 
+if PY2:
+  from collections import Iterable, MutableSet
+else:
+  from collections.abc import Iterable, MutableSet
+
```

Comment:
You can potentially avoid this code if you'd like. Python 3 still includes `Iterable, Mapping`, etc under `collections`, even though it's deprecated and `collections.abc` is preferred.

In Python 3.7, it leads to a deprecation warning that gets printed to console (although maybe we silence that like we do with Pants). In Python 3.8, it will be removed completely, which we're a few years away from that.

--

For Pants, we ultimately decided to just use std lib's `collections` backport—even with 3.8 eventually dropping support—under the logic that Pants won't be using Python 2 by the time 3.8 comes out. 

Because PEX will support Python 2 for a long, long time, I think you're probably right to add the backport.

Reply:
I actually added the backport because I was getting the warning in earlier iterations of the code where I turned on warnings more broadly (for the _pex vendoring re-director code). It felt cheap to ignore it when we already have the `pex.compatibility` module for just this sort of thing.